fdbclient/BackupContainerFileSystem.actor.cpp

```diff
@@ -927,6 +927,9 @@ class BackupContainerFileSystemImpl {
 		// Find the most recent keyrange snapshot through which we can restore filtered key ranges into targetVersion.
 		state std::vector<KeyspaceSnapshotFile> snapshots = wait(bc->listKeyspaceSnapshots());
 		state int i = snapshots.size() - 1;
+		if (targetVersion == earliestVersion) {
+			i = 0;
```

Comment:
This doesn't work with the logic below, i.e., loop at line 933. There we do `i--`, so if 0 is not desirable, we won't try next ones like 1, 2, ... So the change needed is to change the loop to be `i++` if `earliestVersion` is the target version.

Reply:
Both way may not work because the snapshot at head or tail may be incomplete.

So I changed the behavior to explicitly selection the restore version before entering this command based on backup description. PTAL.